packages/core/src/browser/quick-open/quick-input-service.ts

```diff
@@ -63,15 +102,35 @@ export class QuickInputService {
     @inject(QuickOpenService)
     protected readonly quickOpenService: QuickOpenService;
 
+    @inject(ThemeService)
+    protected readonly themeService: ThemeService;
```

Comment:
If you properly encapsulate iconPath handling into the plugin system, you don't need to handle themes, they will be covered via css classes.